lib/synapse/rate_limiter.rb

```diff
@@ -0,0 +1,71 @@
+require 'json'
+
+module Synapse
+  class RateLimiter
+    include Logging
+
+    def initialize(path)
+      @path = path
+
+      # Our virtual clock
+      @time = 0
+
+      # Average at most one restart per minute, with the ability to burst to
+      # two restarts per minute
+      @tokens = 0
+      @max_tokens = 2
```

Comment:
It seems like these should be options? Perhaps for the next iteration, but I could see it being a problem if we effectively only restart once every minute.